### PR TITLE
os: 1.3.1 -> 1.4.0

### PR DESCRIPTION
**Pull request by Benjile**

### CODE REVIEW

File: pkgs/os/os.sh [hunk 1]
```diff
@@ -107,31 +114,62 @@ _print() {
...
     if [[ $ZONE_NAME =~ ^OCB000.* ]]; then
-        INFRA_ZONE_NAME="$(jq -r '.[] | select(.providers.fe.domain_name == env.ZONE_NAME) | .name' < "${CAASCAD_ZONES_FILE}" )"
-        DOMAIN_NAME="$(jq -r '.[] | select(.providers.fe.domain_name == env.ZONE_NAME) | .domain_name' < "${CAASCAD_ZONES_FILE}" )"
+        INFRA_ZONE_NAME="$(jq -e -r '.[] | select(.providers.fe.domain.name == env.ZONE_NAME) | .infra_zone_name' < "${CAASCAD_ZONES_FILE}" )"
+        DOMAIN_NAME="$(jq -r --arg ZONE "${INFRA_ZONE_NAME}" '.|to_entries[] | select(.key == $ZONE) | .value.domain_name' < "${CAASCAD_ZONES_FILE}" )"
     else
         INFRA_ZONE_NAME="$(jq -r '.[env.ZONE_NAME].infra_zone_name' < "${CAASCAD_ZONES_FILE}")"
         DOMAIN_NAME="$(jq -r '.[env.ZONE_NAME].domain_name' < "${CAASCAD_ZONES_FILE}")"
     fi
     export VAULT_ADDR="https://vault.${INFRA_ZONE_NAME}.${DOMAIN_NAME}"
```
Comment (abryko) [inline pkgs/os/os.sh:168]: could also check if INFRA_ZONE_NAME & DOMAIN_NAME are non-null

Comment (Benjile) [reply pkgs/os/os.sh]: done

File: pkgs/os/os.sh [hunk 2]
```diff
@@ -107,31 +144,54 @@ _print() {
           ;;
       esac
     done
```
Comment (abryko) [inline pkgs/os/os.sh:146]: this loop is missing a default case to still consume arguments when they are invalid

File: pkgs/os/os.sh [hunk 3]
```diff
@@ -59,27 +63,61 @@ _refresh() {
...
+    echo "Zone name empty. Use os switch first." >&2
+    exit 1
+  fi
+  echo "${ZONE}"
 }
 
 _parse() {
     if [[ "$#" -eq 0 ]]; then
-        cat "${CURRENT_FILE}";
+        _get_current_zone;
```
Comment (abryko) [inline pkgs/os/os.sh:100]: finding the help is kind of hard, couldn't we display help by default ? 
(don't know if lots of stuff rely on the current behaviour ?)

File: pkgs/os/os.sh [hunk 4]
```diff
@@ -59,27 +63,61 @@ _refresh() {
...
+            _support_template
+            ;;
         refresh|r)
             _refresh;
             ;;
         switch|s)
             if [[ "$#" -eq 2 ]]; then
-                _switch "$2"
+                _switch "${2}"
+                _check_zone "${2}"
```
Comment (abryko) [inline pkgs/os/os.sh:120]: put `_check_zone()` inside of functions fetching or setting zone instead ? (`_switch()` & `_get_current_zone()`)

**Review by abryko — CHANGES_REQUESTED** [review]

File: pkgs/os/os.sh (hunk 2 above)
Comment (Benjile) [reply pkgs/os/os.sh:153]: done

File: pkgs/os/os.sh (hunk 3 above)
Comment (Benjile) [reply pkgs/os/os.sh]: done

File: pkgs/os/os.sh (hunk 4 above)
Comment (Benjile) [reply pkgs/os/os.sh]: done

File: pkgs/os/os.sh [hunk 5]
```diff
@@ -58,28 +62,66 @@ _refresh() {
...
+    echo "${ZONE}" > "${CURRENT_FILE}"
+    _refresh
+    _check_zone "${ZONE}"
+}
+
+_check_zone() {
+  ZONE=${1}
+  if ! jq -e -r --arg ZONE "${ZONE}" '.[] | select(.providers.fe.domain.name == $ZONE) | .infra_zone_name' "${CAASCAD_ZONES_FILE}" &>/dev/null && 
+    ! jq -e -r --arg ZONE "${ZONE}" '.[$ZONE].infra_zone_name' "${CAASCAD_ZONES_FILE}" &>/dev/null; then
+    echo "Zone or domain name ${ZONE} not found" && exit 1;
```
Comment (abryko) [inline pkgs/os/os.sh:75]:
```suggestion
    echo "Zone or domain name ${ZONE} not found" >&2 && exit 1;
```

**Review by abryko — APPROVED** [review]

**Pull request merged**